### DISCUSSION TITLE
Add `twig template` that evaluates to `true` or `false` as an option for conditionals

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#13170] [Ideas]
The recent addition of a Twig template option which outputs a related element ID dynamically for relational condition rules is great!

Would it be possible to have a Twig template option that evaluates to `true` or `false` in conditionals?

I think the existing rules cover a lot (certainly the most common) of use cases, but a condition based on a dynamic boolean value from a twig template would open up possibilities for some crazy stuff. And I mean that in a good way 🙂

**Comment by michaelrog** [comment]
Using object templates as conditions still feels like an anti-pattern. (It's mushy, error-prone, and has unbounded performance implications.) The Condition Builder was created to eliminate the need for this practice in the first place!

That said, for better or worse, it's a common practice in the Craft ecosystem now. If we decide it's too late to be idealistic about this, I would at least reach for _expressions_ rather than _parsed templates_, a la [Symfony Expression Language](https://symfony.com/doc/current/reference/formats/expression_language.html).

**Reply by thupsi** [reply]
Thanks for the input on this! 

I admit that I'm entirely out of my depth when it comes to the actual technical implementation, so my suggestion should only be seen from an end-user-functionality perspective. I'm perfectly willing to accept that it may be a bad idea for all the reasons you mentioned :)

**Reply by brandonkelly** [reply]
We looked into Symfony Expressions for Commerce at one point, but it’s pretty limited compared to Twig expressions.

**Reply by michaelrog** [reply]
> We looked into Symfony Expressions for Commerce at one point, but it’s pretty limited compared to Twig expressions.

Seems like a feature to me. Nudges folks towards using this feature with limited scope — and with "real" expression values, rather than arbitrary string values.

**Reply by piotrpog** [reply]
Don't we use twig code for calculating section url anyway?

**Comment by mmikkel** [comment]
Really needed this today – would be amazing if it happened 🙂

I do get @michaelrog's point, but IMO I don't really see how object template support in conditions would be any more of a worry than having it in something like URI or title formats 🤷

**Reply by brandonkelly** [reply]
At least those settings are exclusively developer-facing, but conditions are also author-facing.

**Reply by mmikkel** [reply]
Ah, right – that's a fair point. But maybe it'd be possible to hide this potential "Template" condition rule type in author-facing condition builders?

**Reply by brandonkelly** [reply]
Yeah, maybe it’s an admin-facing thing only…

**Comment by mmikkel** [comment]
While I still think this feature has merit as described, the "Generated Fields" added in Craft 5.8 can be used in entry conditions, which will cover the use case in a lot of cases.
